common/src/services/userVerification.service.ts

```diff
@@ -19,7 +19,12 @@ export class UserVerificationService implements UserVerificationServiceAbstracti
     async buildRequest<T extends SecretVerificationRequest>(verification: Verification,
         requestClass?: new () => T, alreadyHashed?: boolean) {
         if (verification?.secret == null || verification.secret === '') {
-            throw new Error('No secret provided for verification.');
+            if (verification.type === VerificationType.OTP) {
+                this.handleError(this.i18nService.t('verificationCodeRequired'));
+            } else {
+                this.handleError(this.i18nService.t('masterPasswordRequired'));
+            }
```

Comment:
We duplicate this code with `verifyUser`, can we extract it to a function?

`this.handleError(this.getError(verification.type)`

Reply:
This block has been extracted to `private verifyInput`